### ISSUE TITLE
Potential edge case: percent memory scaling on Windows performance logs

### ISSUE THREAD

**Issue by computergeek125** [nmonvisualizer/nmonvisualizer#109]
If you open a Windows performance log that contains the counter `\\HOSTNAME\Memory\% Committed Bytes In Use`, Nmon Visualizer does not appear to scale the data correctly.

Initially, we thought the counter had gotten nulled because it was shown (correctly) on a scale of 0-100 but had all 0s (opening the csv in excel revealed that this was not the case, it had real data).  At first, I thought the workaround was to postprocess the data to remove the percent sign.  This was not correct, but it showed the real problem: the data was being shown with seemingly the right integer value, but on a scale of `0` to `1E-4` (highest value in the log was `9.5E-5`, which in the raw data was `'95.xx`).

I believe this is being caused by this segment: https://github.com/nmonvisualizer/nmonvisualizer/blob/ead994fb636e78c8c7e636bb84adbfe27433cb53/src/com/ibm/nmon/data/transform/WindowsBytesTransform.java#L37, which appears to be converting a detected counters matching the strings "memory" and "bytes" to megabytes.  (my java is a few years rusty, so I'm not 100% certain of my analysis here)

While normally I would agree with this conversion, in this case Windows has already does the scaling for us, so the number is already within a range of 0..100.  Would it be possible to add a check to bypass the bytes~MB conversion if the counter is also a percentage?

**Comment by hpresnall** [comment]
Agree. That regex needs some work. I think it should ignore any `%` at the beginning of the metric name.

Should be able to fix this in the next week.

This older code also has checks for KB and MB and doesn’t convert those. Are there any metrics you see that are in GB or TB now? Will probably add checks for those just in case.

**Comment by computergeek125** [comment]
Windows performance collector hasn't changed those names around much as far as I can tell.

I admittedly haven't done a deep dive into all the available collectors recently, but a quick skim through my Windows 11 Insider box shows that within the `Memory`, `PhysicalDisk`, and `LogicalDisk` the counters largely are still bytes, with a few KB and MB.  Mixture of unit formatting between `KBytes`, `MBytes`, and `Megabytes`, so per usual Microsoft, Windows is not always internally consistent (that last `Megabytes` shows up on `PhysicalDisk` as `Free Megabytes`).  Even XHCI (USB 3.x) and QUIC (HTTP3) is in byte units.  Other categories not flagged in `VALID_TYPES` may display their units in shorthand, as in `(MB)`

Some of the Storage Spaces move their `%` marker to the end of the counter string name, but they don't appear to also have the edge case of saying `%` and `bytes` at the same time - so this would probably (correctly) be skipped by this code segment even if Storage Spaces categories were in `VALID_TYPES`

**Comment by hpresnall** [comment]
Updated the regex. Please build this and let me know if everything looks ok.

**Comment by computergeek125** [comment]
Sorry for the delay!

I just re-tested a known-problematic data set that exhibited this issue on the 2024-04-04 build and 2025-01-01 build.  The issue is resolved and shows the correct commit values in the newer comple.  Thanks for the speedy triage of this issue!
